pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -574,31 +651,41 @@ func getVolumesFromPodDir(podDir string) ([]podVolume, error) {
 		}
 		podName := podsDirInfo[i].Name()
 		podDir := path.Join(podDir, podName)
-		volumesDir := path.Join(podDir, options.DefaultKubeletVolumesDirName)
-		volumesDirInfo, err := ioutil.ReadDir(volumesDir)
-		if err != nil {
-			glog.Errorf("Could not read volume directory %q: %v", volumesDir, err)
-			continue
-		}
-		for _, volumeDir := range volumesDirInfo {
-			pluginName := volumeDir.Name()
-			volumePluginPath := path.Join(volumesDir, pluginName)
-
-			volumePluginDirs, err := utilfile.ReadDirNoStat(volumePluginPath)
+		// Find both filesystem volume and block volume information
+		// ex. filesystem volume: /pods/{podUid}}/volume/{escapeQualifiedPluginName}/{volumeName}
+		//     block volume:      /pods/{podUid}}/volumeDevices/{escapeQualifiedPluginName}/{volumeName}
+		volumesDirs := []string{path.Join(podDir, options.DefaultKubeletVolumesDirName),
+			path.Join(podDir, options.DefaultKubeletVolumeDevicesDirName)}
+		for _, volumesDir := range volumesDirs {
```

Comment:
can volumesDir be a map with key = volumeMode, and value = path?

Reply:
done.